src/fu-tool.c

```diff
@@ -753,6 +753,80 @@ fu_util_install (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_update (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) devices = NULL;
+
+	/* load engine */
+	if (!fu_util_start_engine (priv, error))
+		return FALSE;
+
+	devices = fu_engine_get_devices (priv->engine, error);
+	for (guint i = 0; i < devices->len; i++) {
+		FwupdDevice *dev = g_ptr_array_index (devices, i);
+		FwupdRelease *rel;
+		const gchar *device_id;
+		const gchar *remote_id;
+		const gchar *uri_tmp;
+		g_autoptr(GPtrArray) rels = NULL;
+		g_autoptr(GError) error_local = NULL;
+
+		if (!fu_util_is_interesting_device (dev))
+			continue;
+		/* only show stuff that has metadata available */
+		if (!fwupd_device_has_flag (dev, FWUPD_DEVICE_FLAG_SUPPORTED))
+			continue;
+
+		device_id = fu_device_get_id (dev);
+		rels = fu_engine_get_upgrades (priv->engine, device_id, &error_local);
+		if (rels == NULL) {
+			g_printerr ("%s\n", error_local->message);
+			continue;
+		}
+
+		rel = g_ptr_array_index (rels, 0);
+		uri_tmp = fwupd_release_get_uri (rel);
+		remote_id = fwupd_release_get_remote_id (rel);
+		if (remote_id != NULL) {
+			FwupdRemote *remote;
+			g_autoptr(GPtrArray) remotes = NULL;
+			g_auto(GStrv) argv = g_new0 (gchar *, 2);
+
+			remotes = fu_engine_get_remotes (priv->engine, error);
```

Comment:
I still think you want `fu_engine_get_remote_by_id()` :)

Reply:
I just saw that earlier comment now.  I'll fixup.